src/Benchmarks/RegexEngineDetectionBenchmarks.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using BenchmarkDotNet.Attributes;
+using Microsoft.Security.Utilities;
+
+namespace Benchmarks
+{
+    public class RegexEngineDetectionBenchmarks
+    {
+        [Benchmark]
+        public void CachedDotNetRegex()
+        {
+            ScanTestExamples(null);
+        }
+
+        [Benchmark]
+        public void RE2()
+        {
+            ScanTestExamples(RE2RegexEngine.Instance);
+        }
+
+        private static void ScanTestExamples(IRegexEngine? regexEngine)
+        {
+            int globalCount = 0;
+
+            // Setting 'generateCorrelatingIds' to false vs. true may be interesting in profiling.
+            var masker = new SecretMasker(WellKnownRegexPatterns.HighConfidenceSecurityModels,
+                                           generateCorrelatingIds: true,
+                                           regexEngine);
+
+            for (int i = 1; i <= 200; i++)
```

Comment:
>200 [](http://example.com/codeflow?start=33&length=3)

This # of iterations produces 8200 true positives, every one of which will be detacted or masked depending on the scenario.